File: src/src_core/fmg/core/img/PolarLightBkTransformer.java
package fmg.core.img;

import fmg.common.HSV;

/** Transforming of background color (rotation of background color) */
public class PolarLightBkTransformer implements IModelTransformer {

   @Override
   public void execute(int currentFrame, int totalFrames, IImageModel model) {
      if (!(model instanceof AnimatedImageModel))
         throw new RuntimeException("Illegal usage transformer");

      AnimatedImageModel am = (AnimatedImageModel)model;
      if (!am.isPolarLights())
         return;

      double rotateAngleDelta = 360.0 / totalFrames; // 360° / TotalFrames
      if (!am.getAnimeDirection())
         rotateAngleDelta = -rotateAngleDelta;

      HSV hsv = new HSV(am.getBackgroundColor());
      hsv.h += rotateAngleDelta;
      am.setBackgroundColor(hsv.toColor());
   }

}

File: src/src_core/fmg/core/img/PolarLightBkTransformer.cs
using System;
using fmg.common;

namespace fmg.core.img {

   /// <summary> Transforming of background color (rotation of background color) </summary>
   public class PolarLightBkTransformer : IModelTransformer {

      public void Execute(IAnimatedModel model) {
         if (!(model is AnimatedImageModel am))
            throw new Exception("Illegal usage transformer");

         if (!am.PolarLights)
            return;

         double rotateAngleDelta = 360.0 / am.TotalFrames; // 360° / TotalFrames
         if (!am.AnimeDirection)
            rotateAngleDelta = -rotateAngleDelta;

         HSV hsv = new HSV(am.BackgroundColor);
         hsv.h += rotateAngleDelta;
         am.BackgroundColor = hsv.ToColor();
      }

   }

}
